### PR TITLE
Enable ESLint

### PR DESCRIPTION
**Pull request by backevik**
- Enable arm64 support
- Add make target for eslint (so should be picked up by pipeline)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,6 +17,9 @@ include tools/git-verify-nodiff/rules.mk
 include tools/golangci-lint/rules.mk
 include tools/goreview/rules.mk
 include tools/semantic-release/rules.mk
+include tools/eslint/rules.mk
+
+example_gen_path := examples/proto/gen/typescript/einride/example/freight/v1
```
Comment (odsod) [inline Makefile:22]: since this is only used in one place I would actually inline it to make things faster to read without jumping around

Comment (backevik) [reply Makefile]: Yup I agree! 🚀

File: tools/eslint/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  extends: ["plugin:@einride/default"],
+    rules: {
+    "jest/no-deprecated-functions": 0,
```
Comment (ericwenn) [inline tools/eslint/.eslintrc.js:4]: Why is this needed? `jest` is not used.

File: examples/proto/gen/typescript/einride/example/syntax/v2/index.ts [hunk 3]
```diff
@@ -1,171 +1,172 @@
 // Code generated by protoc-gen-typescript-http. DO NOT EDIT.
+/* eslint-disable camelcase */
```
Comment (ericwenn) [inline examples/proto/gen/typescript/einride/example/syntax/v2/index.ts:2]: Is it the method names that should be camelcase?
If that is the case I think we could make a breaking change now, and remove this disable.

File: Makefile [hunk 4]
```diff
@@ -44,6 +46,10 @@ $(protoc_gen_typescript_http):
 	$(info [$@] building protoc-gen-typescript-http...)
 	@go build -o $@ .
 
+.PHONY: $(eslint)
+eslint: $(eslint)
+	$(info [$@] linting typescript files...)
+	$(eslint) --config $(eslint_cwd)/.eslintrc.js --quiet "examples/proto/gen/typescript/einride/example/freight/v1/index.ts"
```
Comment (ericwenn) [inline Makefile:52]: Can we run eslint on the entire generated typescript dir instead?

File: tools/eslint/.eslintrc.js (hunk 2 above)
Comment (backevik) [reply tools/eslint/.eslintrc.js:4]: I know. The ESLint plugin however requires it: https://github.com/einride/eslint-plugin/blob/14dcb0b45a75a2dcf1d219aeafcb49f122c7af44/lib/index.js#L8

So I disabled it here. I think we should look into making our ESLint plugin a bit more modular with `peerDependencies`.

File: Makefile (hunk 4 above)
Comment (ericwenn) [reply Makefile:52]: Something like `"examples/proto/gen/typescript/**/*.ts"`

File: tools/eslint/.eslintrc.js (hunk 2 above)
Comment (ericwenn) [reply tools/eslint/.eslintrc.js:4]: So we would need to install `jest` so that the linter can run?

File: examples/proto/gen/typescript/einride/example/syntax/v2/index.ts (hunk 3 above)
Comment (backevik) [reply examples/proto/gen/typescript/einride/example/syntax/v2/index.ts:2]: No, In this case, it's the enum `googletype_LatLng`

File: tools/eslint/.eslintrc.js (hunk 2 above)
Comment (backevik) [reply tools/eslint/.eslintrc.js:4]: Yeah...
Here's the output if you remove this disable: `Unable to detect Jest version - please ensure jest package is installed`.

File: Makefile (hunk 4 above)
Comment (backevik) [reply Makefile:52]: Yeah good point! will update 👍

File: examples/proto/gen/typescript/einride/example/syntax/v2/index.ts (hunk 3 above)
Comment (ericwenn) [reply examples/proto/gen/typescript/einride/example/syntax/v2/index.ts:2]: gotcha! then it is better to ignore.
For context, `_` is used to ensure no duplicate identifiers.

File: Makefile (hunk 4 above)
Comment (backevik) [reply Makefile]: Updated and fixed all lint issues! There was a few edge cases with enums (e.g no leading pipe if only 1 enum or ending the last enum with a semi colon). I'm not sure my solution is the best but it works 🙈

**Review by ericwenn — APPROVED** [review]
